point.js

```diff
@@ -1,110 +1,109 @@
-/*
+/**
  * Point Utils
  * A point is an array of two numbers e.g. [0, 0].
+ *
+ * @flow
  */
 
-var _ = require("underscore");
-var kvector = require("./vector.js");
-var knumber = require("./number.js");
+import _ from "underscore";
 
-var kpoint = {
+import * as kvector from "./vector.js";
+import * as knumber from "./number.js";
 
-    // Rotate point (around origin unless a center is specified)
-    rotateRad: function(point, theta, center) {
-        if (center === undefined) {
-            return kvector.rotateRad(point, theta);
-        } else {
-            return kvector.add(
-                center,
-                kvector.rotateRad(
-                    kvector.subtract(point, center),
-                    theta
-                )
-            );
-        }
-    },
+// A point, in 2D, 3D, or nD space.
+export type Point = $ReadOnlyArray<number>;
```

Comment:
It would be nice to have separate types for these since a lot of the functions that accept points assume that their args are 2D points.   This should be possible, see https://flow.org/try/#0PTAEAEDMBsHsHcBQAXAngBwKagAqwJYB2yATKALygDahArgLYBGmATgDSh1OsC6A3CgzY8RZADkKoACQAlTAEMAJgHlC0VAEEWLeagA8XZiwB8AxNEzJQkWLEkAKdAC5cBYmICUFY6ADeiRABfMwsrdBIXEWIySioSDgBmfgCbWEcSDwEgA.

Reply:
I like that idea.  I think I'd like to do a whole pass (future PR) that makes it explicit that this library deals only with 2D space. I think, like you say, that it's implicitly that, but because `vector` deals with n-ary arrays of numbers, that bleeds out to the functions that sit on top of `vector` even though those functions expect 2D points.